bukuserver/server.py

```diff
@@ -80,6 +81,22 @@ def init_locale(app, context_processor=lambda: {}):
         app.context_processor(lambda: {'lang': '', **context_processor()})
 
 
+# handling popup= URL argument
+def before_request():
+    _post_popup = request.headers.get('Content-Type') != 'application/json' and request.data.get('popup')
+    flask.g.popup = request.args.get('popup') or _post_popup
```

Comment:
Detecting `popup=True` on every request.

The `Content-Type` check is necessary since `request.data.get()` appears to break API requests when POST data is in JSON format.